fineract-progressive-loan/src/test/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanScheduleGeneratorTest.java

```diff
@@ -87,7 +71,7 @@ void testGenerateLoanSchedule() {
 
         ScheduledDateGenerator mockScheduledDateGenerator = Mockito.mock(ScheduledDateGenerator.class);
         ProgressiveLoanScheduleGenerator generator = new ProgressiveLoanScheduleGenerator(mockScheduledDateGenerator, emiCalculator);
-        when(mockScheduledDateGenerator.generateRepaymentPeriods(any(), any(), any())).thenReturn(expectedRepaymentPeriods);
+        when(mockScheduledDateGenerator.generateRepaymentPeriods(any(), any(), any(), any())).thenReturn(expectedRepaymentPeriods);
```

Comment:
Please dont do any mocks... we should be able to generate the loan schedule without any mocking!

Reply:
Fixed.